lib/bolt/cli.rb

```diff
@@ -109,7 +111,10 @@ def create_option_parser(results)
           '* protocol is `ssh` by default, may be `ssh` or `winrm`',
           '* port is `22` by default for SSH, `5985` for winrm (Optional)'
         ) do |nodes|
-          results[:nodes] = parse_nodes(nodes)
+          parsed = parse_nodes(nodes)
+          if parsed
+            results[:nodes] += parsed
+          end
```

Comment:
If we pass `--nodes a --nodes a` then we'll get duplicates, which would regress on https://github.com/puppetlabs/bolt/commit/555b4ed95e86b231bd92d293c5ab702792778658. Also `parse_nodes` always returns an array (possible empty), so checking `if parsed` isn't needed.

Reply:
correct, but we are ensuring Uniq in a secondary PR that would do away with that is a separate issue can and can exist even without the secondary nodes flag.